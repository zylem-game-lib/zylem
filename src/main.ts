import * as THREE from 'three';
import * as RAPIER from '@dimforge/rapier3d-compat';
import { Howl } from 'howler';

import { game, IGameOptions, stage, StageOptions } from './lib/core';

import { PerspectiveType, Perspectives } from './lib/interfaces/perspective';
import { Vect3 } from './lib/core/utility';

import * as actions from './lib/behaviors/actions';
import * as entities from './lib/entities/index';

const Util = {
	...actions
};

const Zylem = {
	Util
};

const { box, sphere, sprite, plane, zone, actor, vessel, ZylemBox } = entities;

namespace Zylem { };

export {
	game,
	stage,
	box,
	sphere,
	sprite,
	plane,
	zone,
	actor,
	vessel,
	actions,
	Perspectives,
	Zylem,
	ZylemBox,
	Howl,
	THREE,
	RAPIER
};
export type { IGameOptions as ZylemGame, StageOptions as ZylemStage, Vect3, PerspectiveType };
