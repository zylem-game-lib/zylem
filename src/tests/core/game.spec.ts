import { describe, expect, it } from 'vitest';
import { game, stage } from '../../main';

describe('create a basic game', () => {
	it('default configuration', () => {
		const result = game();
		expect(result).toMatchSnapshot();
	});
	it('should create a game with multiple stages', async () => {
		const result = game(
			stage(),
			stage(),
			stage()
		);
		expect(result.options.length).toEqual(3);
	});
});
