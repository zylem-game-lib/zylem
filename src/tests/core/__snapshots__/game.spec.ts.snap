// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`create a basic game > default configuration 1`] = `
Game {
  "gameRef": null,
  "options": [],
  "setupOverride": [Function],
  "updateOverride": [Function],
}
`;
